shell/browser/mac/in_app_purchase_observer.mm

```diff
@@ -94,6 +94,25 @@ - (NSString*)dateToISOString:(NSDate*)date {
   return [dateFormatter stringFromDate:date];
 }
 
+/**
+ * Convert a SKPaymentDiscount object to a PaymentDiscount structure.
+ *
+ * @param paymentDiscount - The SKPaymentDiscount object to convert.
+ */
+- (in_app_purchase::PaymentDiscount)skPaymentDiscountToStruct:
+    (SKPaymentDiscount*)paymentDiscount API_AVAILABLE(macosx(10.14.4)) {
+  in_app_purchase::PaymentDiscount paymentDiscountStruct;
+
+  paymentDiscountStruct.identifier = [paymentDiscount.identifier UTF8String];
+  paymentDiscountStruct.keyIdentifier =
+      [paymentDiscount.keyIdentifier UTF8String];
+  // paymentDiscountStruct.nonce = [paymentDiscount.nonce uuidString];
```

Comment:
try `-[NSUUID UUIDString]`: https://developer.apple.com/documentation/foundation/nsuuid/1416585-uuidstring?language=objc
```suggestion
  paymentDiscountStruct.nonce = [paymentDiscount.nonce UUIDString];
```

Reply:
Just pushed the commit 👍